modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/SharedAssetResourceImpl.java

```diff
@@ -18,4 +41,113 @@
 	scope = ServiceScope.PROTOTYPE, service = SharedAssetResource.class
 )
 public class SharedAssetResourceImpl extends BaseSharedAssetResourceImpl {
+
+	@Override
+	public Page<SharedAsset> getMyUserAccountSharedAssetsSharedByMePage(
+			String search, Aggregation aggregation, Filter filter,
+			Pagination pagination, Sort[] sorts)
+		throws Exception {
+
+		return SearchUtil.search(
+			new HashMap<>(), _getSharedByMeBooleanQueryUnsafeConsumer(), filter,
+			SharingEntry.class.getName(), search, pagination,
+			queryConfig -> queryConfig.setSelectedFieldNames("sharingEntryId"),
+			searchContext -> {
+				searchContext.addVulcanAggregation(aggregation);
+				searchContext.setCompanyId(contextCompany.getCompanyId());
+			},
+			sorts,
+			document -> _toSharedAsset(
+				_sharingEntryService.getSharingEntry(
+					GetterUtil.getLong(document.get("sharingEntryId")))));
+	}
+
+	@Override
+	public Page<SharedAsset> getMyUserAccountSharedAssetsSharedWithMePage(
+			String search, Aggregation aggregation, Filter filter,
+			Pagination pagination, Sort[] sorts)
+		throws Exception {
+
+		return SearchUtil.search(
+			new HashMap<>(), _getSharedWithMeBooleanQueryUnsafeConsumer(),
+			filter, SharingEntry.class.getName(), search, pagination,
+			queryConfig -> queryConfig.setSelectedFieldNames("sharingEntryId"),
+			searchContext -> {
+				searchContext.addVulcanAggregation(aggregation);
+				searchContext.setCompanyId(contextCompany.getCompanyId());
+			},
+			sorts,
+			document -> _toSharedAsset(
+				_sharingEntryService.getSharingEntry(
+					GetterUtil.getLong(document.get("sharingEntryId")))));
+	}
+
+	private UnsafeConsumer<BooleanQuery, Exception>
+		_getSharedByMeBooleanQueryUnsafeConsumer() {
+
+		return booleanQuery -> {
+			BooleanFilter booleanFilter = booleanQuery.getPreBooleanFilter();
+
+			booleanFilter.add(
+				new TermFilter(
+					"userId", String.valueOf(contextUser.getUserId())),
+				BooleanClauseOccur.MUST);
+		};
+	}
+
+	private UnsafeConsumer<BooleanQuery, Exception>
+		_getSharedWithMeBooleanQueryUnsafeConsumer() {
+
+		return booleanQuery -> {
+			BooleanFilter booleanFilter = booleanQuery.getPreBooleanFilter();
+
+			TermsFilter sharedToUserGroupIdTermsFilter = new TermsFilter(
```

Comment:
prefix `shared` is needed?
(same on the other TermsFilter)

Reply:
maybe not, I am changing it